.github/workflows/docs.yml

```diff
@@ -0,0 +1,40 @@
+name: put rust docs on Github Pages
+
+on:
+  push:
+    branches: ["main"]
+  workflow_dispatch:
+
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+concurrency:
+  group: "pages"
+  cancel-in-progress: true
+
+jobs:
+    build:
+        environment:
+            name: github-pages
+            url: ${{ steps.deployment.outputs.page_url }}
+
+        runs-on: ubuntu-latest
+
+        steps:
+            - uses: actions/checkout@v3
+
+            - name: Install riscv64 target
+              run: rustup target add riscv64gc-unknown-none-elf
+            - name: Generate Docs
+              run: make doc
```

Comment:
Can the make doc step from the other flow be used here? Or, maybe the doc generation should be removed from the other flow as to avoid duplicating the work.

Reply:
I think they are really two different things. One tests that the docs can be generated as part of CI, and the other is actually deploying the docs. I could possibly merge the files if that makes sense.

Reply:
also, there are some differences (the environment, some of the actions imported) and it seems that while docs is nice, the rust workflow is essential and not worth messing with.

Reply:
Ok, yes, that is a better way to do it. It's all in one workflow now.